examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -17,19 +17,23 @@ async function pollForServiceUpdates(
 	setExternalData: React.Dispatch<React.SetStateAction<Record<string, unknown>>>,
 ): Promise<void> {
 	try {
-		const response = await fetch(`http://localhost:${externalDataServicePort}/fetch-tasks`, {
-			method: "GET",
-			headers: {
-				"Access-Control-Allow-Origin": "*",
-				"Content-Type": "application/json",
+		const taskListId = "task-list-1";
+		const response = await fetch(
+			`http://localhost:${externalDataServicePort}/fetch-tasks/${taskListId}`,
+			{
+				method: "GET",
+				headers: {
+					"Access-Control-Allow-Origin": "*",
+					"Content-Type": "application/json",
+				},
 			},
-		});
+		);
 
 		const responseBody = (await response.json()) as Record<string, unknown>;
-		const newData = responseBody.taskList as TaskData;
-		if (newData !== undefined && !isEqual(newData, externalData)) {
+		const newData = responseBody.taskList as TaskListData;
+		if (newData !== undefined && !isEqual(newData[taskListId], externalData)) {
```

Comment:
Since the request already specifies taskListId in the URL, I think having it again in the response body is unnecessary.